### PR TITLE
netdata: fix darwin; cleanup; 1.47.0 -> 1.47.1

### LINKED ISSUE #341362: Build failure: netdata broken on aarch64-darwin
### Steps To Reproduce

1. build *netdata*

### Build log

```
error: Package ‘lm-sensors-3.6.0’ in /nix/store/88wskw5ssly5xdqy9wi8ax03diqpibmc-source/pkgs/os-specific/linux/lm-sensors/default.nix:61 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
         package.meta.platforms = [
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.

```

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"aarch64-darwin"`
 - host os: `Darwin 23.4.0, macOS 14.4`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.6`
 - nixpkgs: `not found`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes
Resolves https://github.com/NixOS/nixpkgs/issues/341362

Not very familiar with netdata usage, so would like someone to test it that is. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 341472` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>netdataCloud</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netdata</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
> Result of `nixpkgs-review pr 341472` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:
> 
>     * netdataCloud
> 
> 
> 1 package built:
> 
>     * netdata

Doh, forgot to check the cloud version...

File: pkgs/tools/system/netdata/default.nix [hunk 1]
```diff
@@ -1,36 +1,67 @@
...
   version = "1.47.0";
   pname = "netdata";
 
   src = fetchFromGitHub {
     owner = "netdata";
     repo = "netdata";
-    rev = "v${version}";
-    hash = if withCloudUi
-      then "sha256-eMapy4HQaEblDfZt2uVSfBWRlkstX7TxZDBgSv5bW/I="
+    rev = "v${finalAttrs.version}}";
```
Comment (sarahec) [inline pkgs/tools/system/netdata/default.nix:58]:
```suggestion
    rev = "v${finalAttrs.version}";
```

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 341472` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

File: pkgs/tools/system/netdata/default.nix (hunk 1 above)
Comment (khaneliman) [reply pkgs/tools/system/netdata/default.nix:58]: lol ty... i hate when my cmp does that...

Comment (sarahec) [reply pkgs/tools/system/netdata/default.nix:58]: Nice to know I'm not the only one

**Comment by sarahec** [comment]
> oh but, that's on linux. Which shouldn't have changed from this. Is that failing on master atm?

It's the typo in the version number. Couldn't pull version 1.4.7}

Same on Darwin x86. Puzzled it's not happening on Darwin-aarch64.

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 341472` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
> Two requests:
> 
>     1. Consider bumping to 1.47.1 (just released)
> 
>     2. Please squash your commits per [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)

I would like to keep the logical units of operations separate, as I do it most PRs so that no-ops don't need to be reverted if a breaking change happens in the main commit with logical changes.

**Comment by sarahec** [comment]
LGTM, but I'd like an opinion from someone more senior. CC: @NixOS/darwin-maintainers

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 341472` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
> Result of `nixpkgs-review pr 341472` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 2 packages failed to build:

Hmm.. I'll run x86 locally to see what's up.

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 341472` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

**Comment by sarahec** [comment]
Unfortunately, my x86 Darwin nix install has suddenly stopped working. It may be a bit while I'm troubleshooting.

**Comment by khaneliman** [comment]
>     1. Consider bumping to 1.47.1 (just released)

alright, just bumped it after making sure x86 built for me on the last version

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 341472` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 341472` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 341472` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 341472` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>netdataCloud</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netdata</li>
  </ul>
</details>

**Review by Aleksanaa — APPROVED** [review]

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 341472` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>netdata</li>
    <li>netdataCloud</li>
  </ul>
</details>

**Comment by smancill** [comment]
Ah, my build on Darwin had a package failure probably because I have sandbox enabled.

**Comment by sarahec** [comment]
nixpkgs-review is good on x86_64-Linux, x86_64-Darwin, and aarch64-Darwin

**Pull request merged**